### PR TITLE
Fix diagrams, address some of Dan's Comments 

### PR DESCRIPTION
**Pull request by astoycos**
TODO: Rebase after https://github.com/abhiraut/enhancements/pull/12 merges

### CODE REVIEW

File: keps/sig-network/2091-cluster-network-policy/README.md [hunk 1]
```diff
@@ -206,24 +225,22 @@ CNP resources, such that no two CNP instances can be created with the same `prio
 (TODO: Is such admission control viable?)
```
Comment (Dyanngg) [inline keps/sig-network/2091-cluster-network-policy/README.md:225]: This is what we need to discuss in today's meeting. If we do not reach a conclusion, at least we need to move the TODOs into a separate section as Dan suggested

Comment (astoycos) [reply keps/sig-network/2091-cluster-network-policy/README.md:225]: +1 I'm making an out of band list of things I would like to discuss I'll make sure to add this one

File: keps/sig-network/2091-cluster-network-policy/README.md [hunk 2]
```diff
@@ -206,24 +225,22 @@ CNP resources, such that no two CNP instances can be created with the same `prio
...
 For conformance and portability purposes, it will be required that implementations
-support up to 1000 CNP priorities (0 through 999) per cluster and up to 100 rules
-within a CNP instance.  An implementation can optionally choose to support larger numbers
-of priority values and rules, however if an implementation supports a
-smaller number of rules and instances than these recommended ranges, it will be considered
-non-conformant.  This ensures that policy manifests written using priorities 0 through
-999 for instance are portable across clusters and implementations without need to re-write
-policy manifests or re-adjust priority allocations when moving applications from 1
-kubernetes cluster to another.
+support up to 1000 numeric CNP priorities (0 through 999) per cluster and up to 100 rules
```
Comment (astoycos) [inline keps/sig-network/2091-cluster-network-policy/README.md:228]: I think we punt this to the CNIs and leave it undefined

File: keps/sig-network/2091-cluster-network-policy/README.md [hunk 3]
```diff
@@ -232,13 +249,13 @@ relative priority ordering of all configured rules and CNP instances.
 
 #### Rule identifiers
 
-In order to collect traffic statistics corresponding to
-a rule, it is necessary to identify the rule which allows/denies that traffic.
+In order to collect traffic statistics corresponding to a rule in the future,
```
Comment (astoycos) [inline keps/sig-network/2091-cluster-network-policy/README.md:252]: I still think we should remove this mention of "traffic statistics" and instead just argue that it's easier for readability's sake + future proofing

File: keps/sig-network/2091-cluster-network-policy/README.md [hunk 4]
```diff
@@ -612,25 +644,29 @@ to CNP rules.
 #### Sample spec for Story 1: Deny traffic at a cluster level
 
 ```yaml
-apiVersion: netpol.networking.k8s.io/v1alpha1
+apiVersion: policy.networking.k8s.io/v1alpha1
```
Comment (Dyanngg) [inline keps/sig-network/2091-cluster-network-policy/README.md:647]: There was a yaml spec change for this Story in my first commit but seems lost during rebasing?

Comment (astoycos) [reply keps/sig-network/2091-cluster-network-policy/README.md:647]: I think I had an overriding change here... Specifically I changed the sample policy to day "deny all ingress traffic to `sensitive-ns`" before It said the inverse I believe

File: keps/sig-network/2091-admin-network-policy/README.md [hunk 5]
```diff
@@ -362,41 +357,36 @@ providers, as is the case with the NetworkPolicy API.
...
-project for example, could be extended to support ClusterNetworkPolicy. kubectl plugin extensions
-that allow listing of policies and rules in this manner should be developed. Similarly
-a kubectl plugin extension should be developed that checks whether communication is allowed
-between a specified pair of source and destination pods for traffic of a specified protocol/ port type.
+affect traffic between workloads might not be very straightforward.
+
+To mitigate this risk and improve usability, some additional in-tree tooling
+for both the Admin and Developer will need to be created. For the Admin, it is 
+safe to assume they will have the correct RBAC roles to list all the NetworkPolicies 
+and AdminNetworkPolicies in a cluster.  Therefore, the Admin oriented tooling should
```
Comment (Dyanngg) [inline keps/sig-network/2091-admin-network-policy/README.md:377]: extra space before "Therefore"

File: keps/sig-network/2091-admin-network-policy/README.md [hunk 6]
```diff
@@ -362,41 +357,36 @@ providers, as is the case with the NetworkPolicy API.
...
-a kubectl plugin extension should be developed that checks whether communication is allowed
-between a specified pair of source and destination pods for traffic of a specified protocol/ port type.
+affect traffic between workloads might not be very straightforward.
+
+To mitigate this risk and improve usability, some additional in-tree tooling
+for both the Admin and Developer will need to be created. For the Admin, it is 
+safe to assume they will have the correct RBAC roles to list all the NetworkPolicies 
+and AdminNetworkPolicies in a cluster.  Therefore, the Admin oriented tooling should 
+be able to both alert the Admin to any overriding of NetworkPolicies that may occur for a given 
+AdminNetworkPolicy and provide a warning if there is another ANP with the same
```
Comment (Dyanngg) [inline keps/sig-network/2091-admin-network-policy/README.md:379]: what about "any overriding of NetworkPolicies that may occur if a new AdminNetworkPolicy is to be created"

File: keps/sig-network/2091-admin-network-policy/README.md [hunk 7]
```diff
@@ -408,164 +398,186 @@ also like to consider the following set of proposals as future work items:
...
+  // If this field is present and contains at least one item, then this rule 
+  // allows/denies/passes traffic only if the traffic matches at least one port in the list.
+  // +optional
+  Ports     []networkingv1.NetworkPolicyPort
+	// List of sources from which traffic will be allowed/denied/passed to the entities 
+  // selected by this AdminNetworkPolicyRule. Items in this list are combined using a logical OR 
+  // operation. If this field is empty or missing, this rule matches no sources. 
+  // If this field is present and contains at least one item, this rule
+	// allows/denies/passes traffic from the defined AdminNetworkPolicyPeer
+	From      []AdminNetworkPolicyPeer
```
Comment (Dyanngg) [inline keps/sig-network/2091-admin-network-policy/README.md:476]: Indent

File: keps/sig-network/2091-admin-network-policy/README.md [hunk 8]
```diff
@@ -408,164 +398,186 @@ also like to consider the following set of proposals as future work items:
...
-// Namespaces define a way to select Namespaces in the cluster.
-type Namespaces struct {
-	Scope       NamespaceMatchType
-	// Labels should be set only when scope is "SameLabels".
-	Labels      []string
-	// Selector should be set only when scope is "Selector".
-	// Namespaces.Selector has the same effect as NamespaceSelector in NetworkPolicy.
-	Selector    *metav1.LabelSelector
+// NamespaceSetAndPodSelector defines a way to select Namespaces and Pods in a cluster 
+type NamespaceSetAndPod struct {
```
Comment (Dyanngg) [inline keps/sig-network/2091-admin-network-policy/README.md:534]: s/NamespaceSetAndPod/NamespaceSetAndPodSelector/g

File: keps/sig-network/2091-admin-network-policy/README.md [hunk 9]
```diff
@@ -204,66 +192,63 @@ that needs to be allowed for certain components to work as expected (egress to
...
-The relative precedence of the rules within a CNP will be determined by the order in which the
+The relative precedence of the rules within a ANP will be determined by the order in which the
 rule is written. Thus, a rule that appears at the top of the ingress/egress rules would take the
 highest precedence.
 
 Conflict resolution: Two policies are considered to be conflicting if they are assigned the
-same `priority`. In order to avoid such conflicts, we propose to include validation for
-CNP resources, such that no two CNP instances can be created with the same `priority`.
-(TODO: Is such admission control viable?)
+same `priority` and select the same resources or a union of resources. In order to avoid such conflicts,
```
[the remark this answers is not in the document]
Comment (astoycos) [reply keps/sig-network/2091-admin-network-policy/README.md:217]: Agreed, fixed

File: keps/sig-network/2091-admin-network-policy/README.md [hunk 10]
```diff
@@ -362,41 +357,36 @@ providers, as is the case with the NetworkPolicy API.
...
-particular Pod more challenging than when there's only NetworkPolicy resources.
-
-In case where a ClusterNetworkPolicy `Pass` rule,
-ClusterNetworkPolicy `Deny` rule, ClusterNetworkPolicy `Allow` rule,
+To understand why traffic between a pair of Pods is allowed or denied, a list of 
+NetworkPolicy resources in both Pods' Namespace used to be sufficient (considering 
+no other CRDs in the cluster tries to alter traffic behavior). With the introduction
+of AdminNetworkPolicy, all users could face some difficulties determining why 
+NetworkPolicies did not take effect even if they know to look for, 
+and have the correct permissions to list, any interfering AdminNetworkPolicy rules.
```
[the remark this answers is not in the document]
Comment (astoycos) [reply keps/sig-network/2091-admin-network-policy/README.md:365]: Yeah This one was a bit chopped up, let me rephrase

File: keps/sig-network/2091-admin-network-policy/README.md (hunk 6 above)
Comment (astoycos) [reply keps/sig-network/2091-admin-network-policy/README.md:379]: Done

File: keps/sig-network/2091-admin-network-policy/README.md (hunk 7 above)
Comment (astoycos) [reply keps/sig-network/2091-admin-network-policy/README.md:476]: Weird it renders correctly in my vscode :), fixed

File: keps/sig-network/2091-admin-network-policy/README.md (hunk 8 above)
Comment (astoycos) [reply keps/sig-network/2091-admin-network-policy/README.md:534]: Fixed

**Comment by astoycos** [comment]
Thanks for the review! @Dyanngg

File: keps/sig-network/2091-admin-network-policy/README.md [hunk 11]
```diff
@@ -0,0 +1,1117 @@
...
+)
+
+// Exactly one of the selector pointers should be set in a given AdminNetworkPolicyPeer
+type AdminNetworkPolicyPeer struct {
+  NamespaceSetSelector         *NamespaceSet
+  NamespaceSetAndPodSelector   *NamespaceSetAndPod
+}
+
+// NamespaceSet defines a flexible way to select Namespaces in a cluster.
+type NamespaceSet struct {
```
Comment (Dyanngg) [inline keps/sig-network/2091-admin-network-policy/README.md:523]: I'm thinking if `NamespaceScope` might be a better name for this struct. And we could change the `Scope` field down below to something like `MatchBy`, so it will read "matchBy: Selector/Self/NotSelf....". But we should not block this PR from merging for this... Let's see if others can come up with better naming here

File: keps/sig-network/2091-admin-network-policy/README.md [hunk 12]
```diff
@@ -0,0 +1,1117 @@
...
+  name: cluster-wide-allow-example
+spec:
+  priority: 30
+  - appliedTo:
+     namespaceSelector: {}
+    ingress:
+     - action: Allow
+       from:
+       - namespaceSetSelector: 
+         scope: Selector
```
Comment (Dyanngg) [inline keps/sig-network/2091-admin-network-policy/README.md:688]: Wrong indent

File: keps/sig-network/2091-admin-network-policy/README.md [hunk 13]
```diff
@@ -0,0 +1,1117 @@
...
+         namespaceSelector:
+          matchLabels:
+            kubernetes.io/metadata.name: monitoring-ns
+    egress: 
+      - action: Allow
+        to: 
+        - namespaceSetAndPodSelector:
+          scope: Selector
```
Comment (Dyanngg) [inline keps/sig-network/2091-admin-network-policy/README.md:696]: ditto

File: keps/sig-network/2091-admin-network-policy/README.md [hunk 14]
```diff
@@ -0,0 +1,1117 @@
...
+  name: pub-svc-delegate-example
+spec:
+  priority: 20
+  - appliedTo:
+      namespaceSelector: {}
+    egress:
+    - action: Pass
+      to:
+      - namespaceSetAndPodSelector:
+        scope: Selector
```
Comment (Dyanngg) [inline keps/sig-network/2091-admin-network-policy/README.md:720]: ditto

File: keps/sig-network/2091-admin-network-policy/README.md [hunk 15]
```diff
@@ -0,0 +1,1117 @@
...
+metadata: 
+  name: baseline-rule-example
+spec:
+  priority: 0
+  - appliedTo:
+     namespaceSelector: {}
+    ingress:
+     - action: Deny   # zero-trust cluster default security posture
+       from:
+       - NamespaceSetSelector
```
Comment (Dyanngg) [inline keps/sig-network/2091-admin-network-policy/README.md:795]: missing ":" and wrong indent for the next line

File: keps/sig-network/2091-admin-network-policy/README.md [hunk 16]
```diff
@@ -0,0 +1,1117 @@
...
+     - action: Pass
+       from:
+       - namespaceSetSelector:
+          scope: SameLabels
+          labels:
+            - tenant
+     - action: Deny   # Deny everything else other than same tenant traffic
+       from:
+       - namespaceSetSelector:
+         scope: Selector
```
Comment (Dyanngg) [inline keps/sig-network/2091-admin-network-policy/README.md:771]: ditto

File: keps/sig-network/2091-admin-network-policy/README.md (hunk 11 above)
Comment (astoycos) [reply keps/sig-network/2091-admin-network-policy/README.md:523]: +1 I agree happy to change this in a follow up

File: keps/sig-network/2091-admin-network-policy/README.md (hunk 15 above)
Comment (astoycos) [reply keps/sig-network/2091-admin-network-policy/README.md:795]: Fixed
